sdk/core/Azure.Core.TestFramework/src/VersionString.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core.TestFramework
+{
+    internal readonly struct VersionString : IComparable
```

Comment:
Is this still needed?

Reply:
I need somewhere to encaspulate the parsing / comparison of a version string

Reply:
Can we move it to where it is used as part of TestVersion?

Reply:
Its cleaner to separate this as its own thing.  There is also potential here to share this logic since this isn't the only place we do this, but I am punting on that for now I have a follow up I will track